library_generation/test/generate_library_unit_tests.sh

```diff
@@ -0,0 +1,261 @@
+#!/usr/bin/env bash
+
+set -xeo pipefail
+
+# Variables used to generate final result
+total_num=0
+succeed_num=0
+failed_num=0
+failed_tests=""
+# Unit tests against ./utilities.sh
+script_dir=$(dirname "$(readlink -f "$0")")
+source "${script_dir}"/../utilities.sh
+
+# Helper functions, they shouldn't be called outside this file.
+__assertEquals() {
+  expected=$1
```

Comment:
go/shell-style#use-local-variables
Functions should declare their local variables with the `local` keyword. We have several functions (not only in this PR) around that are not following this style. cc: @blakeli0 @suztomo would it make sense to have a "styling up" PR to make our scripts _more_ compliant with the guidelines?

Reply:
go/shell-style#constants-environment-variables-and-readonly-variables
Another example, to have constants with upper case (e.g. `SCRIPT_DIR` instead of `script_dir`)

Reply:
nvm, I saw at the end of the changes we have https://github.com/googleapis/sdk-platform-java/pull/1948/files#r1311724425 to address this!

Reply:
I plan to propose a PR to format code after merging this one.